src/app/map/Chat/ChatAIIntegration.tsx

```
@@ -0,0 +1,112 @@
+import type { ChatMessage } from './types'
+
+/**
+ * Component that handles AI integration for the chat
+ * This is mounted inside the ChatPanel to enable AI responses
+ */
+export function ChatAIIntegration() {
+  const chatState = useChatState()
+  // Try to get cache context - might not be available in tests
+  let cacheState: ReturnType<typeof useMapCacheContext>['state'] | null = null
+  try {
+    const context = useMapCacheContext()
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)